### PR TITLE
Implement frame counter

### PR DESCRIPTION
**Pull request by eucan**
This is a nice debugging feature. Compositor sends information about received number of frames from a surface

### CODE REVIEW

File: weston-ivi-shell/src/ivi-controller.c [hunk 1]
```diff
@@ -61,8 +61,10 @@ struct ivisurface {
     const struct ivi_layout_surface_properties *prop;
     struct wl_listener property_changed;
     struct wl_listener surface_destroy_listener;
+    struct wl_listener commited;
```
Comment (mtey) [inline weston-ivi-shell/src/ivi-controller.c]: typo: committed

File: weston-ivi-shell/src/ivi-controller.c [hunk 2]
```diff
@@ -577,20 +579,24 @@ send_surface_stats(struct ivicontroller *ctrl,
...
+
     /* Get pid that creates surface */
     surface = lyt->surface_get_weston_surface(layout_surface);
 
     target_client = wl_resource_get_client(surface->resource);
 
     wl_client_get_credentials(target_client, &pid, &uid, &gid);
 
-    ivi_wm_send_surface_stats(ctrl->resource, surface_id, 0, 0, 0, pid, "");
+
```
Comment (mtey) [inline weston-ivi-shell/src/ivi-controller.c]: Please remove additional blank line

File: weston-ivi-shell/src/ivi-controller.c [hunk 3]
```diff
@@ -1561,6 +1567,15 @@ create_layer(struct ivishell *shell,
     return ivilayer;
 }
 
+static void
+surface_commited(struct wl_listener *listener, void *data)
```
Comment (mtey) [inline weston-ivi-shell/src/ivi-controller.c]: typo: committed

File: weston-ivi-shell/src/ivi-controller.c [hunk 4]
```diff
@@ -1561,6 +1567,15 @@ create_layer(struct ivishell *shell,
...
+{
+    struct ivisurface *ivisurf = wl_container_of(listener, ivisurf, commited);
```
Comment (mtey) [inline weston-ivi-shell/src/ivi-controller.c]: typo: committed

File: weston-ivi-shell/src/ivi-controller.c [hunk 5]
```diff
@@ -1582,6 +1598,10 @@ create_surface(struct ivishell *shell,
     wl_list_insert(&shell->list_surface, &ivisurf->link);
     wl_list_init(&ivisurf->notification_list);
 
+    ivisurf->commited.notify = surface_commited;
```
Comment (mtey) [inline weston-ivi-shell/src/ivi-controller.c]: typo: committed

File: weston-ivi-shell/src/ivi-controller.c [hunk 6]
```diff
@@ -1582,6 +1598,10 @@ create_surface(struct ivishell *shell,
...
+    surface = lyt->surface_get_weston_surface(layout_surface);
+    wl_signal_add(&surface->commit_signal, &ivisurf->commited);
```
Comment (mtey) [inline weston-ivi-shell/src/ivi-controller.c]: commit_signal is correct, but &ivisurf->committed

File: weston-ivi-shell/src/ivi-controller.c [hunk 7]
```diff
@@ -1697,6 +1717,7 @@ surface_event_remove(struct wl_listener *listener, void *data)
 
     wl_list_remove(&ivisurf->link);
     wl_list_remove(&ivisurf->property_changed.link);
+    wl_list_remove(&ivisurf->commited.link);
```
Comment (mtey) [inline weston-ivi-shell/src/ivi-controller.c]: typo: committed

**Comment by efriedrich** [comment]
thanks for the review, even there are only minor findings I would like to get them fix and since @eucan is not available, we will create a new pull request out of it.
thanks @eucan, thanks @mtey

**Pull request closed without merging**